### PR TITLE
[Enhancement] netbeans - force interactive shell when limited detection

### PR DESCRIPTION
**Pull request by cordoval**
| Q | A |
| --- | --- |
| Bug Fix? | y |
| New Feature? | n |
| BC Breaks? | n |
| Deprecations? | n |
| Tests Pass? | y |
| Fixed Tickets | #9946 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Console/Application.php [hunk 1]
```diff
@@ -855,7 +855,7 @@ protected function configureIO(InputInterface $input, OutputInterface $output)
             $input->setInteractive(false);
         } elseif (function_exists('posix_isatty') && $this->getHelperSet()->has('dialog')) {
             $inputStream = $this->getHelperSet()->get('dialog')->getInputStream();
-            if (!@posix_isatty($inputStream)) {
+            if (!@posix_isatty($inputStream) && !(true === getenv('SHELL_INTERACTIVE'))) {
```
Comment (xabbuh) [inline src/Symfony/Component/Console/Application.php]: should be `true !== getenv('SHELL_INTERACTIVE')`

**Comment by cordoval** [comment]
fabbot.io is now queued? how to reset that one?

File: src/Symfony/Component/Console/Application.php [hunk 2]
```diff
@@ -855,7 +855,7 @@ protected function configureIO(InputInterface $input, OutputInterface $output)
             $input->setInteractive(false);
         } elseif (function_exists('posix_isatty') && $this->getHelperSet()->has('dialog')) {
             $inputStream = $this->getHelperSet()->get('dialog')->getInputStream();
-            if (!@posix_isatty($inputStream)) {
+            if (!@posix_isatty($inputStream) && true !== getenv('SHELL_INTERACTIVE')) {
```
Comment (stof) [inline src/Symfony/Component/Console/Application.php]: ``getenv`` can return a string or ``false`` when the env var is not set. But it cannot be ``true``

**Comment by cordoval** [comment]
done @stof

**Comment by cordoval** [comment]
@tmysik can you please help me verify with this that the terminal now remains interactive? thanks

**Comment by cordoval** [comment]
I think the failure in tests is unrelated to this PR.

**Comment by tmysik** [comment]
@cordoval Will be glad to do that! However, could you please provide steps how can I get Symfony with the fix/change? Thanks.

**Comment by cordoval** [comment]
It should be simple, you need to add my fork git remote add cordoval git@github.com:cordoval/symfony.git
then checkout fix-netbeans-interactive-shell branch and you should be in there to test.

**Comment by tmysik** [comment]
@cordoval I must be doing something wrong but not sure what :/ I get:

```
Fatal error: Call to undefined method Symfony\Component\ClassLoader\DebugUniversalClassLoader::useIncludePath() in /home/gapon/NetBeansProjects/symfony2nb/vendor/symfony/src/Symfony/Component/ClassLoader/DebugUniversalClassLoader.php on line 41
```

The command NetBeans runs is:

```
"/usr/bin/php" "/home/gapon/NetBeansProjects/symfony2nb/app/console" "--ansi" "list" "--xml"
```

Thanks.

**Comment by cordoval** [comment]
you have to run composer and all

**Comment by tmysik** [comment]
@cordoval Sorry, no idea what you mean exactly. What I did:

* created new Symfony project as stated on the download page (using ```symfony``` installer)
* replaced the Symfony sources by your checkout (in ```vendor``` directory)
* run Symfony command to list commands

The result is:

```
Fatal error: Class 'Symfony\Bundle\DebugBundle\DebugBundle' not found in /home/gapon/NetBeansProjects/s2nb/app/AppKernel.php on line 23
Done.
```

Thanks.

**Comment by cordoval** [comment]
no, don't use the installer. Development wise i always prefer to clone a symfony-standard edition and checkout branch 2.7, and then run composer install --prefer-source, then go under vendor and cd symfony/symfony then add remotes and git remote update and then checkout my branch after adding the remote. Tedious work but it is what is contributing to any package so far.

**Comment by tmysik** [comment]
@cordoval I followed your steps but still got:

```
Fatal error: Class 'Symfony\Bundle\DebugBundle\DebugBundle' not found in /home/gapon/NetBeansProjects/symfony-standard/app/AppKernel.php on line 23
```

Just for verification:

```
gapon@cattie ~/NetBeansProjects/symfony-standard/vendor/symfony/symfony $ git checkout fix-netbeans-interactive-shell 
Branch fix-netbeans-interactive-shell set up to track remote branch fix-netbeans-interactive-shell from cordoval.
Switched to a new branch 'fix-netbeans-interactive-shell'
gapon@cattie ~/NetBeansProjects/symfony-standard/vendor/symfony/symfony $ git branch 
  2.7
* fix-netbeans-interactive-shell
```

Thanks.

**Comment by stof** [comment]
@tmysik the branch with the bugfix is based on symfony 2.3, not on 2.7. So use the 2.4 branch of symfony-standard too to test it

**Comment by tmysik** [comment]
@stof So, I switched symfony-standard to ```2.4``` branch. Went to ```vendor/symfony/symfony``` and switched again to ```fix-netbeans-interactive-shell``` branch. However, listing Symfony commands ends with this error:

```
[RuntimeException]
The console handler requires symfony/monolog-bridge 2.4+
```

Perhaps it would be great if someone could create a ZIP with properly configured symfony-standard project and proper Symfony version for NetBeans problem - I could it then just easily verify in NetBeans :)

Thanks.

**Comment by stof** [comment]
I said 2.3, not 2.4

**Comment by stof** [comment]
ah actually, I mixed the version in my previous comment, sorry

**Comment by tmysik** [comment]
@stof Perfect, thanks for the steps, it works now! And I can confirm that the Symfony commands are interactive again in NetBeans so everything works for us now. Thanks a lot, guys!

**Comment by cordoval** [comment]
yeah sorry to have said 2.7, i really meant 2.3. Thanks @stof. @tmysik nice! great job!

This should be LGTM then :+1:

**Comment by tmysik** [comment]
Thanks a lot for solving this problem!

**Comment by fabpot** [comment]
Thank you @cordoval.

**Pull request closed without merging**